apps/keyboard/js/keyboard/view_manager.js

```diff
@@ -101,7 +104,9 @@ ViewManager.prototype.render = function (layout, flags, callback) {
     };
 
     pageView = this._createPageView(layout, options);
-    this.pageViews.set(pageId, pageView);
+    if (!layout.updateByIMEngine) {
+      this.pageViews.set(pageId, pageView);
```

Comment:
Since this `pageView` is not put into the `pageViews` map, it would be GC'd when it is being removed from `ViewManager#currentPageView`. However, it's DOM element will remain in the DOM and you will leak memory. Please also create a `destroy` method on the page view and call it when we detach it from the view manager.

(I am sorry I did not spot this on previous reviews)